### PR TITLE
S0072 masked modulus10 2

### PR DESCRIPTION
**Pull request by KnowledgeForwardSolutions**
This pull request adds comprehensive support for masked validation and calculation in the `Modulus10_2Algorithm`, including new tests for various masking scenarios and edge cases. It introduces the `IMaskedCheckDigitAlgorithm` interface implementation, adds new mask classes, and significantly expands the unit tests to cover masked and unmasked validation, input edge cases, and real-world examples like IMO numbers.

Key changes include:

**Modulus10_2Algorithm Enhancements:**

* `Modulus10_2Algorithm` now implements the `IMaskedCheckDigitAlgorithm` interface, enabling support for masked check digit validation and calculation.
* Added internal logic to enforce a maximum of 9 processed digits for masked validation scenarios.
* Refactored digit validation to use the `IsInvalidDigit()` extension for clarity and correctness. [[1]](diffhunk://#diff-2001ac93c31d79f5213e87f07de88e5d3851da9de8392bf8949c8882c77a213fL46-R51) [[2]](diffhunk://#diff-2001ac93c31d79f5213e87f07de88e5d3851da9de8392bf8949c8882c77a213fL76-R82)

**Unit Test Additions and Improvements:**

* Added extensive unit tests for masked validation, including:
  - Null, empty, and excessive-length input scenarios.
  - Various masking strategies (`AcceptAllMask`, `ImoNumberCheckDigitMask`, `GroupsOfThreeCheckDigitMask`, `RejectAllMask`).
  - Real-world IMO number validation and error detection.
* Expanded tests for unmasked validation and calculation, including new edge cases for non-digit characters and benchmark values.
* Added new mask classes (`ImoNumberCheckDigitMask`) and test helpers for more flexible test coverage.
* Added property tests for the new `MaskedAlgorithms.Modulus10_2` property.

**Benchmark Test Adjustments:**

* Updated `ValidateMaskedArguments` in `NumericAlgorithmBenchmarks.cs` to focus on `Modulus10_2` and comment out other algorithms, aligning with the new test focus.

These changes improve both the robustness and flexibility of the `Modulus10_2Algorithm` and ensure thorough test coverage for masked and unmasked scenarios.

### CODE REVIEW

File: CheckDigits.Net/GeneralAlgorithms/Modulus10_2Algorithm.cs [hunk 1]
```diff
@@ -43,10 +44,11 @@ public Boolean TryCalculateCheckDigit(String value, out Char checkDigit)
...
       var t = 0;
-      for (var index = 0; index < value.Length; index++)
+      var processLength = value.Length;
+      for (var index = 0; index < processLength; index++)
       {
          var currentDigit = value![index].ToIntegerDigit();
-         if (currentDigit < 0 || currentDigit > 9)
+         if (currentDigit.IsInvalidDigit())
          {
                return false;
```
[the remark this answers is not in the document]
Comment (KnowledgeForwardSolutions) [reply CheckDigits.Net/GeneralAlgorithms/Modulus10_2Algorithm.cs]: fixed

File: Documentation/Stories/Backlog/S0073-MaskedModulus10_1.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# S0073: MaskedModulus10_1
+
+## Story
+**As a** .NET developer  
+**I want** to use the Modulus10_1 algorithm with a check digit mask to handle values that have been formatted for human readablility.
```
[the remark this answers is not in the document]
Comment (KnowledgeForwardSolutions) [reply Documentation/Stories/Backlog/S0073-MaskedModulus10_1.md:5]: ignored

File: Documentation/Stories/Backlog/S0074-MaskedAlphanumericMod97_10.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# S0074: MaskedAlphanumericMod97_10
+
+## Story
+**As a** .NET developer  
+**I want** to use the AlphanumericMod97_10 algorithm with a check digit mask to handle values that have been formatted for human readablility.
```
[the remark this answers is not in the document]
Comment (KnowledgeForwardSolutions) [reply Documentation/Stories/Backlog/S0074-MaskedAlphanumericMod97_10.md:5]: ignored

File: CheckDigits.Net.Tests.Unit/GeneralAlgorithms/Modulus10_2AlgorithmTests.cs [hunk 4]
```diff
@@ -2,170 +2,315 @@
...
+   [InlineData("9706692")]       // IMO Number 9707792 with twin error 77 -> 66
+   public void Modulus10_2Algorithm_Validate_ShouldReturnFalse_WhenInputContainsDetectableError(String value)
+      => _sut.Validate(value).Should().BeFalse();
+
+   [Fact]
+   public void Modulus10_2Algorithm_Validate_ShouldReturnTrue_WhenInputIsAllZeros()
+      => _sut.Validate("0000000000").Should().BeTrue();
+
+   [Fact]
+   public void ModulusAlgorithm_Validate_ShouldReturnTrue_WhenCheckDigitIsCalculatesAsZero()
```
[the remark this answers is not in the document]
Comment (KnowledgeForwardSolutions) [reply CheckDigits.Net.Tests.Unit/GeneralAlgorithms/Modulus10_2AlgorithmTests.cs]: fixed

File: CheckDigits.Net.Tests.Unit/GeneralAlgorithms/Modulus10_2AlgorithmTests.cs [hunk 5]
```diff
@@ -2,170 +2,315 @@
...
+   [InlineData("IMO9706692")]       // IMO Number 9707792 with twin error 77 -> 66
+   public void Modulus10_2Algorithm_ValidateMasked_ShouldReturnFalse_WhenInputContainsDetectableError(String value)
+      => _sut.Validate(value, _imoNumberMask).Should().BeFalse();
+
+   [Fact]
+   public void Modulus10_2Algorithm_ValidateMasked_ShouldReturnTrue_WhenInputIsAllZeros()
+      => _sut.Validate("000 000 000 0", _groupsOfThreeMask).Should().BeTrue();
+
+   [Fact]
+   public void ModulusAlgorithm_ValidateMasked_ShouldReturnTrue_WhenCheckDigitIsCalculatesAsZero()
```
[the remark this answers is not in the document]
Comment (KnowledgeForwardSolutions) [reply CheckDigits.Net.Tests.Unit/GeneralAlgorithms/Modulus10_2AlgorithmTests.cs]: fixed

File: CheckDigits.Net.Tests.Benchmarks/NumericAlgorithmBenchmarks.cs [hunk 6]
```diff
@@ -211,61 +211,65 @@ public IEnumerable<Object[]> ValidateArguments()
...
+      yield return [Algorithms.Modulus10_2, Algorithms.Modulus10_2.AlgorithmName, "140 662 7"];
+      yield return [Algorithms.Modulus10_2, Algorithms.Modulus10_2.AlgorithmName, "140 662 538 9"];
+
+      //yield return [Algorithms.Modulus11_27Decimal, Algorithms.Modulus11_27Decimal.AlgorithmName, "140 6"];
+      //yield return [Algorithms.Modulus11_27Decimal, Algorithms.Modulus11_27Decimal.AlgorithmName, "140 662 0"];
+      //yield return [Algorithms.Modulus11_27Decimal, Algorithms.Modulus11_27Decimal.AlgorithmName, "140 662 538 5"];
+      //yield return [Algorithms.Modulus11_27Decimal, Algorithms.Modulus11_27Decimal.AlgorithmName, "140 662 538 042 1"];
+      //yield return [Algorithms.Modulus11_27Decimal, Algorithms.Modulus11_27Decimal.AlgorithmName, "140 662 538 042 551 0"];
+      //yield return [Algorithms.Modulus11_27Decimal, Algorithms.Modulus11_27Decimal.AlgorithmName, "140 662 538 042 551 028 8"];
+      //yield return [Algorithms.Modulus11_27Decimal, Algorithms.Modulus11_27Decimal.AlgorithmName, "140 662 538 042 551 028 265 0"];
```
[the remark this answers is not in the document]
Comment (KnowledgeForwardSolutions) [reply CheckDigits.Net.Tests.Benchmarks/NumericAlgorithmBenchmarks.cs:248]: ignored
